File: iOS/PengChauFerry/HolidayDao.swift
//
//  HolidayDao.swift
//  PengChauFerry
//
//  Created by Lionello Lunesu on 7/3/2020.
//  Copyright © 2020 Lionello Lunesu. All rights reserved.
//

import Foundation

class HolidayDao {
    private var set = Set<String>()

    func query(today: LocalDate) -> Bool {
        let day = today.toString()
//        return db.readableDatabase.query(
//            DbOpenHelper.HOLIDAYS,
//            arrayOf("date"),
//            "date=?",
//            arrayOf(day),
//            null,
//            null,
//            null
//        ).use {
//            it.moveToNext()
//        }
        return set.contains(day)
    }

    func save(holidays: Array<LocalDate>) {
        holidays.forEach {
            insert(date: $0)
        }
    }

    func insert(date: LocalDate) {
        let day = date.toString()
        set.insert(day)
//        val contentValues = ContentValues()
//        contentValues.put("date", day)
//        db.writableDatabase.insertWithOnConflict(DbOpenHelper.HOLIDAYS, null, contentValues, CONFLICT_IGNORE)
    }

    func delete(date: LocalDate) {
        let day = date.toString()
        set.remove(day)
//        db.writableDatabase.delete(DbOpenHelper.HOLIDAYS, "date=?", arrayOf(day))
    }

}

File: Android/app/src/main/java/com/lunesu/pengchauferry/HolidayDao.kt
package com.lunesu.pengchauferry

import android.content.ContentValues
import android.database.sqlite.SQLiteDatabase.CONFLICT_IGNORE
import org.joda.time.LocalDate

class HolidayDao(private val db: DbOpenHelper) {

    fun query(today: LocalDate): Boolean {
        return db.readableDatabase.query(
            DbOpenHelper.HOLIDAYS,
            arrayOf("date"),
            "date=?",
            arrayOf(today.toString()),
            null,
            null,
            null
        ).use {
            it.moveToNext()
        }
    }

    fun save(holidays: List<LocalDate>) {
        holidays.forEach {
            insert(it)
        }
    }

    fun insert(date: LocalDate) {
        val day = date.toString()
        val contentValues = ContentValues()
        contentValues.put("date", day)
        db.writableDatabase.insertWithOnConflict(DbOpenHelper.HOLIDAYS, null, contentValues, CONFLICT_IGNORE)
    }

    fun delete(date: LocalDate) {
        val day = date.toString()
        db.writableDatabase.delete(DbOpenHelper.HOLIDAYS, "date=?", arrayOf(day))
    }
}
